File: python/56.merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        intervals.sort(key=lambda x: x[0])
        res = [intervals[0]]
        
        for i in intervals[1:]:
            if i[0] > res[-1][1]:
                res.append(i)
            else:
                res[-1][1] = max(res[-1][1], i[1])

        return res

File: golang/56.merge-intervals.go
import (
    "sort"
)

func merge(intervals [][]int) [][]int {
    n := len(intervals)

    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })

    res := [][]int{intervals[0]}

    for i := 1; i < n; i++ {
        cur := intervals[i]
        last := len(res) - 1

        if last >= 0 && res[last][1] >= cur[0] {
            res[last][1] = max(res[last][1], cur[1])
        } else {
            res = append(res, cur)
        }
    }

    return res
}